amethyst_assets/src/storage.rs

```diff
@@ -37,6 +37,18 @@ pub struct AssetStorage<A: Asset> {
     pub(crate) processed: Arc<MsQueue<Processed<A>>>,
     reloads: Vec<(WeakHandle<A>, Box<Reload<A>>)>,
     unused_handles: MsQueue<Handle<A>>,
+    //requeue: Vec<Processed<A>>,
```

Comment:
If anyone know how I can make this work with Send + Sync, feel free to shoot me a question. The basics are that I want to save the allocation for assets that need reprocessing. Right now it just allocates a new Vec every frame, which is stupid. The problem is that adding the above makes `AssetStorage` non Send Sync. Should I just unsafe impl Send + Sync, since I know that `AssetStorage` can only be accessed mutably at one location?

Reply:
No, the asset storage is too complex to get such an impl. I'm not able to review this currently.

Reply:
Do you have an alternative? I'd like to avoid doing an allocation every frame. I know for a fact that the `Vec` will only be used from a single location which is not called in parallel, so how do I tell rust not to worry about it ?

Reply:
Wrapping it in `Mutex` works, I can use `get_mut` and it won't be locked, and the compiler accepts `AssetStorage` as `Send + Sync` still.